### PR TITLE
Makin clientid and encryptionkey mandatory in encrypted share

### LINKED ISSUE #357: for encrypted share, `clientid` and `encryptionpublickey` should be mandatory
when try to download an encrypted file, I get this error: 

```
0box-sdk        [DEBUG]  2023/02/19 21:16:20.546326 filemetaworker.go:74: File meta response status: 200 OK
0box-sdk        [INFO]   2023/02/19 21:16:20.551954 filemetaworker.go:73: File Meta result:{"actual_file_hash":"774590317b96727aaa484a1087f1f444361b001eb32a2b4c3ca221fb25f437a7","actual_file_size":120752,"actual_thumbnail_hash":"","actual_thumbnail_size":0,"allocation_id":"fa346fd2cde360df79198ab09bdd3188c646469b35274fbcfdecde1588eb91f0","chunk_size":65536,"commit_meta_txns":[],"content_hash":"a98cd73a4bfaad06cd3e995c01bcaa709fe466ee37fbe4ebc76a81723e96b918","created_at":1676837451,"custom_meta":"","encrypted_key":"t9Kwc5W4muQEVIZ3kJk6eqWOzbdK3A3bdk/cRH9ni44=","hash":"a4b31e1de8cf273b87fe1d6cdae409b038fa3eaa41d41f5f30bfab118ce1ce41","lookup_hash":"e2d1f02690fc33ad0533dfbad0547fd16f43a7d9c609d9f97f12c176a5395830","merkle_root":"7dd532ae21ac5896e7aee0b699b8fb268f6fa6e06ecb6a2e625f93929a8d6c39","mimetype":"application/octet-stream","name":"go4.sum","num_of_blocks":2,"path_hash":"e2d1f02690fc33ad0533dfbad0547fd16f43a7d9c609d9f97f12c176a5395830","size":121296,"thumbnail_hash":"","thumbnail_size":0,"type":"f","updated_at":1676837451}
0box-sdk        [DEBUG]  2023/02/19 21:16:20.552149 filemetaworker.go:74: File meta response status: 200 OK
0box-sdk        [INFO]   2023/02/19 21:16:20.552884 downloadworker.go:339: Downloading file with size: 120752 from start block: 0 and end block: 2. Actual size per blobber: 131072
 0 / 120752 [----------------------------------------------------------------------------------------------------------------------------------------]   0.00%0box-sdk        [INFO]   2023/02/19 21:16:20.570672 downloadworker.go:399: Downloading block 1 - 2
0box-sdk        [ERROR]  2023/02/19 21:16:20.732247 downloadworker.go:198: Unsuccessful download. Error: response_error: invalid Ed25519 curve point
0box-sdk        [ERROR]  2023/02/19 21:16:20.773446 downloadworker.go:198: Unsuccessful download. Error: response_error: invalid Ed25519 curve point
 0 / 120752 [-------------------------------------------------------------------------------------------------------------------------------------]   0.00% 0s
Error in file operation: Download failed for block 1.
download_failed: 2 failed blobbers exceeded 1 remaining blobbers. Download errors: Error Unsuccessful download. Error: response_error: invalid Ed25519 curve point from http://100.78.70.98:5055 Error Unsuccessful download. Error: response_error: invalid Ed25519 curve point from http://100.78.70.98:5056
```

### PR DESCRIPTION
**Pull request by SinanEkm**
A brief description of the changes in this PR:

Related issue:  [zboxcli#357](https://github.com/0chain/zboxcli/issues/357)

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zboxcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- 0chain: 
- blobber:
- gosdk:
- system_test:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: cmd/share.go [hunk 1]
```diff
@@ -40,7 +40,18 @@ var shareCmd = &cobra.Command{
...
+		}
+
+		fileMeta, err := allocationObj.GetFileMeta(remotepath)
+		if err != nil {
+			PrintError("Error in getting the file meta of the object." + err.Error())
+			os.Exit(1)
+		}
+
+		if len(fileMeta.EncryptedKey) > 0 && (fflags.Changed("clientid") == false || fflags.Changed("encryptionpublickey") == false) {
+			PrintError("Clientid and/or encryptionpublickey are missing for the encrypted share!")
```
Comment (Kishan-Dhakan) [inline cmd/share.go]: This is not ideal, please use persistent flags like how it is done in other commands. Look at other commands' code to see how persistent flags work.

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]

**Review by Kishan-Dhakan — APPROVED** [review]

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
clientid should be checked in the if condition too

**Comment by SinanEkm** [comment]
> clientid should be checked in the if condition too

I know that but I realized that clientid flag shouldn't be used if it is public share. I tried to use clientid flag by leaving it blank but it didn't work like that, I think clientid flag shouldn't use at all for public share. So, that's why I excluded clientid flag. @Kishan-Dhakan

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/TestShareFile  |
| gosdk  | staging  |
| zboxcli  | 357-for-encrypted-share-clientid-and-encryptionpublickey-should-be-mandatory  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | fix/TestShareFile  |
| gosdk  | staging  |
| zboxcli  | 357-for-encrypted-share-clientid-and-encryptionpublickey-should-be-mandatory  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**